djangocms_navigation/models.py

```diff
@@ -0,0 +1,58 @@
+from django.conf import settings
+from django.contrib.contenttypes.fields import GenericForeignKey
+from django.contrib.contenttypes.models import ContentType
+from django.contrib.sites.models import Site
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+from treebeard.mp_tree import MP_Node
+
+
+class Menu(models.Model):
+    """
+    MenuContent Grouper
+    """
+    pass
+
+
+class MenuContent(models.Model):
+    title = models.CharField(verbose_name=_('title'), max_length=100)
+    site = models.ForeignKey(
+        Site,
+        on_delete=models.PROTECT,
+    )
+    menu = models.OneToOneField(
+        Menu,
+        on_delete=models.CASCADE,
+        primary_key=True,
```

Comment:
so if it's an FK it can't be a PK anymore.
Btw, not something you can use here (because it shouldn't really be a one2one), but for future reference, when you have this kind of db structure, this is a pretty cool django feature: https://docs.djangoproject.com/en/2.1/topics/db/models/#multi-table-inheritance